connect/src/main/java/com/stripe/android/connect/webview/StripeDownloadManager.kt

```diff
@@ -0,0 +1,62 @@
+package com.stripe.android.connect.webview
+
+import android.app.DownloadManager
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import android.webkit.URLUtil
+
+/**
+ * Provides an interface for various download and file operations. Useful for mocking in tests.
+ */
+internal interface StripeDownloadManager {
+    /**
+     * Returns the ID of the download, or null if the download could not be started.
+     */
+    fun enqueue(request: DownloadManager.Request): Long?
+
+    /**
+     * Returns a [DownloadManager.Request] for the given URI.
+     */
+    fun getDownloadRequest(uri: String): DownloadManager.Request
+
+    /**
+     * Returns a best-guess file name for the given URL, content disposition, and MIME type.
+     */
+    fun getFileName(url: String, contentDisposition: String? = null, mimeType: String? = null): String
```

Comment:
same with this one, with both of these it seems like the end result is just that you want the file name, and i'm wondering if you could just return that along with the ID when you start the DL request